control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -24,15 +24,15 @@ type KubeAPIServerImages struct {
 }
 
 type KubeAPIServerParams struct {
-	APIServer           configv1.APIServer          `json:"apiServer"`
-	Authentication      configv1.Authentication     `json:"authentication"`
-	FeatureGate         configv1.FeatureGate        `json:"featureGate"`
-	Network             configv1.Network            `json:"network"`
-	OAuth               configv1.OAuth              `json:"oauth"`
-	Image               configv1.Image              `json:"image"`
-	Scheduler           configv1.Scheduler          `json:"scheduler"`
-	CloudProvider       string                      `json:"cloudProvider"`
-	CloudProviderConfig corev1.LocalObjectReference `json:"cloudProviderConfig"`
+	APIServer           configv1.APIServer           `json:"apiServer"`
+	Authentication      configv1.Authentication      `json:"authentication"`
+	FeatureGate         configv1.FeatureGate         `json:"featureGate"`
+	Network             configv1.Network             `json:"network"`
+	OAuth               configv1.OAuth               `json:"oauth"`
+	Image               configv1.Image               `json:"image"`
+	Scheduler           configv1.Scheduler           `json:"scheduler"`
+	CloudProvider       string                       `json:"cloudProvider"`
+	CloudProviderConfig *corev1.LocalObjectReference `json:"cloudProviderConfig"`
```

Comment:
why isn't this a field having  or embedding `KubeAPIServerConfigParams` just like `config.DeploymentConfig`?

Reply:
One purpose of this struct is to help deserialize global config that may have been provided by end users. That's the reason it includes things like `Image` and `Scheduler` which correspond to OCP global config resources. KubeAPIServerConfigParams is extracting the relevant parts of all that config that is only applicable to the KAS config. DeploymentConfig is its own struct because it's meant to be applied as a whole to any deployment rendered by the control plane operator.